File: banda/IvanRC-banda-Java/src/main/java/modelsBandaMusicos/Percusionista.java
package modelsBandaMusicos;

public class Percusionista extends Musico implements IPercusionista {

    String nombre;
    Double salario = 1500.00;
    int añosExperiencia;
    tipoPercusion tipoDePercusion;

    public Percusionista(String nombre, int añosExperiencia, tipoPercusion tipoDePercusion) {
        super(nombre, añosExperiencia);
        this.tipoDePercusion = tipoDePercusion;
    }

    public tipoPercusion getTipoDePercusion() {
        return tipoDePercusion;
    }

    public double recalcularSalario(double salario) {
        return IPercusionista.super.recalcularSalario(salario);
    }
}

File: banda/MarioBueno_Banda_Kotlin/src/main/kotlin/models/Percusionista.kt
//Mario Bueno López
//maarioo2525@gmail.com

package models

import interfaces.IPercusionista
import kotlin.math.round

class Percusionista (override var nombre: String?, override var experiencia: Int, private var tipoDePercusion: String): Musico(nombre, experiencia), IPercusionista {
    override fun respirar() {
        println("Soy $nombre y estoy respirando.")
    }
    override fun interpretar() {
        println("Soy $nombre, músico profesional durante $experiencia años, y estoy interpretando")
    }
    override fun tocarPercusion() {
        println("Estoy tocando $tipoDePercusion")
    }
    companion object {
        var salario: Double = round(Musico.salario * 1.35)
    }
}